lib/aptible/cli/subcommands/logs.rb

```diff
@@ -25,6 +28,149 @@ def logs
+                 'AWS account. You must provide your AWS credentials via ' \
+                 'the environment variables AWS_ACCESS_KEY_ID and ' \
+                 'AWS_SECRET_ACCESS_KEY'
+
+            # Required to retrieve files
+            option :region,
+                   desc: 'The AWS region your S3 bucket resides in',
+                   type: :string, required: true
+            option :bucket,
+                   desc: 'The name of your S3 bucket',
+                   type: :string, required: true
+            option :stack,
+                   desc: 'The name of the Stack to download logs from',
+                   type: :string, required: true
+            option :decryption_keys,
+                   desc: 'The Aptible-provided keys for decryption. ' \
+                         '(Comma separated if multiple)',
+                   type: :array, required: true
+
+            # For identifying files to download
+            option :string_matches,
+                   desc: 'The strings to match in log file names.',
+                   type: :array
+            option :app_id,
+                   desc: 'The Application ID to download logs for.',
+                   type: :numeric
+            option :database_id,
+                   desc: 'The Database ID to download logs for.',
+                   type: :numeric
+            option :proxy_id,
+                   desc: 'The Endpoint ID to download logs for.',
+                   type: :numeric
+            option :container_id,
+                   desc: 'The full (64 char) container ID to download logs for'
+            option :start_date,
+                   desc: 'Get logs starting from this (GMT) date ' \
+                         '(format: YYYY-MM-DD)',
+                   type: :string
+            option :end_date,
+                   desc: 'Get logs before this (GMT) date (format: YYYY-MM-DD)',
+                   type: :string
+
+            # We don't download by default
+            option :download_location,
+                   desc: 'The local path place downloaded log files. ' \
+                         'If you do not set this option, the file names ' \
+                         'will be shown, but not downloaded.',
+                   type: :string
+
+            def logs_from_archive
+              ensure_aws_creds
+              validate_log_search_options(options)
+
+              id_options = [
+                options[:app_id],
+                options[:database_id],
+                options[:proxy_id],
+                options[:container_id]
+              ]
+
+              date_options = [options[:start_date], options[:end_date]]
+
+              r_type = 'apps' if options[:app_id]
+              r_type = 'databases' if options[:database_id]
+              r_type = 'proxy' if options[:proxy_id]
+
+              if date_options.any?
+                begin
+                  start_date = gmt_date(options[:start_date])
```

Comment:
if these are strings, can't we just compare these with equalities? why do we care what dates they are as long as they are inbound in the same timezone/format (is this not always the case?)

`"a" < "b" == true` for example, but with dates

Reply:
I hadn't originally used YYYY-MM-DD on the input, deferring just to `Time.parse(user_input)`. Also I wasn't certain if the YYY-MM-DD format was going to pass UX review, but this code can handle date comparison no matter the input format, as well as if we allow them to entire time as well -- we'll just need to update the stprtime, and all the comparison will still work.

Also, I chose "GMT EVERYWHERE" because the filenames will have times in GMT, and the JSON log timestamps are in GMT, so there will be no discrepancy between what you type when you search, and the time of/in the files.

Reply:
👍 i'm totally cool with us using time parses for future proofing as well if that's how we feel. your logic makes sense, i just referenced it as sometimes as a developer working on times, it's a relief when using ISOs that support string comparisons (strings are so much easier than time things)